python/pyiceberg/catalog/hive.py

```diff
@@ -235,8 +235,8 @@ def identifier_to_database_and_table(
 
         return tuple_identifier[0], tuple_identifier[1]
 
-    def __init__(self, name: str, properties: Properties, uri: str):
-        super().__init__(name, properties)
+    def __init__(self, name: str, uri: str, **properties: str):
+        super().__init__(name, **properties)
```

Comment:
We will probably want to move everything into properties, like Pydantic does. I don't think it makes much sense to make the caller pull out certain properties, like `uri`.

Reply:
I'm not sure if I fully understand. Right now you can pass the `uri` in as a positional argument `RestCatalog(name, uri)` and also a keyword argument `RestCatalog(name, uri=uri)`. This will automatically be mapped to `uri`. It will also raise an error when it isn't passed in. We can also go full keyword arguments (which is what a lot of Pythonistas recommend because it is more readable), but I think it is nice to define the uri explicitly because it provides autocompletion in the IDE.